File: ruby/bob/bob.rb
module Bob
  module_function

  def hey(input)
    silent   = input.strip.empty?
    question = input.rstrip.end_with? '?'
    # contains at least 1 letter, but no lower case letters
    shouting = input =~ /^(?=.*[[:alpha:]])(?!.*[[:lower:]])/

    if    question && shouting then "Calm down, I know what I'm doing!"
    elsif question             then 'Sure.'
    elsif shouting             then 'Whoa, chill out!'
    elsif silent               then 'Fine. Be that way!'
    else                            'Whatever.'
    end
  end
end

File: nim/bob/bob.nim
import strutils


proc hey*(phrase: string): string =
  if phrase.isEmptyOrWhitespace:
    return "Fine. Be that way!"

  var shouting = phrase.contains({'A'..'Z'}) and not phrase.contains({'a'..'z'})
  var asking   = phrase.strip.endsWith('?')

  if shouting and asking:
    return "Calm down, I know what I'm doing!"
  elif shouting:
    return "Whoa, chill out!"
  elif asking:
    return "Sure."
  else:
    return "Whatever."
